src/topic/topic_collection.go

```diff
@@ -41,7 +42,7 @@ func StartTopicPool(poolSize int, wg *sync.WaitGroup, client connection.Client)
 }
 
 // GetTopics retrieves the list of topics to collect based on the user-provided configuration
-func GetTopics(client connection.Client) ([]string, error) {
+func GetTopics(client connection.Client, zkConn zookeeper.Connection) ([]string, error) {
```

Comment:
Rather than having a function with both arguments, could we have a `type topicGetter func() ([]string, error)` and pass it to this function? We then could build two functions, `zookeeperTopicGetter()` and `brokerTopicGetter`.